r/tests/testthat/test-RecordBatch.R

```diff
@@ -519,15 +519,57 @@ test_that("ARROW-11769 - grouping preserved in record batch creation", {
 })
 
 test_that("ARROW-12729 - length returns number of columns in RecordBatch", {
-  
+
   tbl <- tibble::tibble(
     int = 1:10,
     fct = factor(rep(c("A", "B"), 5)),
     fct2 = factor(rep(c("C", "D"), each = 5)),
   )
-  
+
   rb <- record_batch(!!!tbl)
-  
+
   expect_identical(length(rb), 3L)
-  
-})
\ No newline at end of file
+
+})
+
+test_that("RecordBatchReader to C-interface", {
+  tab <- Table$create(example_data)
+
+  # the new way, with a scan -> RecordBatchReader
+  stream_ptr <- allocate_arrow_array_stream()
+  on.exit(delete_arrow_array_stream(stream_ptr))
+  scan <- Scanner$create(tab)
+  reader <- scan$ToRecordBatchReader()
+  reader$export_to_c(stream_ptr)
+  circle <- RecordBatchStreamReader$import_from_c(stream_ptr)
+  tab_from_c_new <- circle$read_table()
+  expect_equal(tab, tab_from_c_new)
+
+  # the new way, with RecordBatchStreamReader
+  stream_ptr_new <- allocate_arrow_array_stream()
+  on.exit(delete_arrow_array_stream(stream_ptr_new))
+  bytes <- write_to_raw(example_data)
+  expect_type(bytes, "raw")
+  reader_new <- RecordBatchStreamReader$create(bytes)
+  reader_new$export_to_c(stream_ptr_new)
+  circle_new <- RecordBatchStreamReader$import_from_c(stream_ptr_new)
+  tab_from_c_new <- circle_new$read_table()
+  expect_equal(tab, tab_from_c_new)
+})
+
+test_that("RecordBatch to C-interface", {
+  batch <- RecordBatch$create(example_data)
+
+  # the new way, with an array
```

Comment:
how is this "new"?

Reply:
This is cruft from part of my testing process, to confirm I was actually testing what I thought I was actually testing. I should have removed this and the others already